File: Introductory Problems/number spiral/number.py
def number_spiral(x, y): 
    largest = max(x, y)
    smallest = min(x, y)
    value = largest ** 2
    b = 1
    while b < largest:
        value -= 1
        b = b + 1
        if x > y and b == smallest:
            print(value);
            return
    while largest > smallest:
        value -= 1
        largest = largest - 1
    print(value);

number_spiral(3, 2)

File: Introductory Problems/number spiral/number.js
function numberSpiral(x, y) {
  let max = Math.max(x, y);
  let min = Math.min(x, y);
  let value = max ** 2;
  let b = 1;
  while (b < max) {
    value -= 1;
    b++;
    if (x > y && b == min) {
      console.log(value);
      return;
    }
  }
  while (max > min) {
    value -= 1;
    max--;
  }
  console.log(value);
}

numberSpiral(5, 3);
